src/wallet/wallet.cpp

```diff
@@ -2435,9 +2435,6 @@ bool CWallet::CommitTransaction(CWalletTx& wtxNew, CReserveKey& reservekey)
                 coin.BindWallet(this);
                 NotifyTransactionChanged(this, coin.GetHash(), CT_UPDATED);
             }
-
-            if (fFileBacked)
```

Comment:
IMO the non-file-backend state is only used in UnitTests, right? We should also consider dropping it completely (in a later PR).